core/arch/instr_shared.c

```diff
@@ -1832,6 +1832,39 @@ instr_reads_from_reg(instr_t *instr, reg_id_t reg, dr_opnd_query_flags_t flags)
     return false;
 }
 
+/* in this func, it must be the exact same register, not a sub reg. ie. eax!=ax */
+bool
+instr_reads_from_exact_reg(instr_t *instr, reg_id_t reg, dr_opnd_query_flags_t flags)
+{
+    int i;
+    opnd_t opnd;
+
+    if (!TEST(DR_QUERY_INCLUDE_COND_SRCS, flags) && instr_is_predicated(instr) &&
+        !instr_predicate_reads_srcs(instr_get_predicate(instr)))
+        return false;
+
+    for (i = 0; i < instr_num_srcs(instr); i++) {
```

Comment:
I think we want to exclude OP_nop_modrm just like instr_reads_from_reg() does (unfortunately hard to just omit from IR).

Reply:
Done